src/providers/autocomplete/AutocompleteStrategy.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
```

Comment:
Same as the sources, I'd like a new directory where each strategy has its own files